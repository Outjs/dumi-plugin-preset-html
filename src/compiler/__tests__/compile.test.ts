import { describe, expect, test } from 'vitest';
import { compiler } from '../node';

const { compileSFC } = compiler;

describe('Html compilation test', () => {
  test('script', () => {
    expect(
      compileSFC({
        id: 'xx',
        code: `
        <script>
        console.log('hello');
        </script>
      `,
        filename: 'xx.html',
      }),
    ).toMatchSnapshot();
  });
  test('script + html', () => {
    expect(
      compileSFC({
        id: 'xx',
        code: `
        <script>
          const msg = 'Hello!'
          function log() {
            console.log(msg)
          }
        </script>
        <button @click="log">button</button>
      `,
        filename: 'xx.html',
      }),
    ).toMatchSnapshot();
  });

  test('two script tags', () => {
    expect(
      compileSFC({
        id: 'xx',
        code: `
        <script>
          const msg = 'Hello!'
          function log() {
            console.log(msg)
          }
        </script>
        <script>
        console.log('hello again');
        </script>
        <button @click="log">button</button>
      `,
        filename: 'xx.html',
      }),
    ).toMatchSnapshot();
  });

  test('script + html + style', () => {
    expect(
      compileSFC({
        id: 'xx',
        code: `
        <script>
          const msg = 'Hello!'
          function log() {
            console.log(msg)
          }
        </script>
        <style>
         .example {
            color: red;
          }
        </style>
        <button @click="log" class="example">button</button>
      `,
        filename: 'xx.html',
      }),
    ).toMatchSnapshot();
  });

  test('html only', () => {
    expect(
      compileSFC({
        id: 'xx',
        code: `
        <p>template only</p>
      `,
        filename: 'xx.html',
      }),
    ).toMatchSnapshot();
  });
});
