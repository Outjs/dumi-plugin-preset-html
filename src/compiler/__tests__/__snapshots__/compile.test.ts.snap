// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Html compilation test > script`] = `
{
  "css": "",
  "js": "
async function() {
  const __html__ = {};
__html__.render = function() {
    setTimeout(()=>{
        try {
            console.log('hello');
        } catch (error) {
            console.error(error);
        }
    }, 100);
    return ``;
};
__html__.__id__ = "634c1b1a";
return {
    default: __html__
};

}
`;

exports[`Html compilation test > script + html`] = `
{
  "css": "",
  "js": "
async function() {
  const __html__ = {};
__html__.render = function() {
    setTimeout(()=>{
        try {
            const msg = 'Hello!';
            function log() {
                console.log(msg);
            }
        } catch (error) {
            console.error(error);
        }
    }, 100);
    return `
<button @click="log">button</button>`;
};
__html__.__id__ = "fcaf92d6";
return {
    default: __html__
};

}
`;

